src/clients/mocks/MockSpokePoolClient.ts

```diff
@@ -94,8 +95,12 @@ export class MockSpokePoolClient extends SpokePoolClient {
   generateDeposit(deposit: DepositWithBlock): Event {
     const event = "FundsDeposited";
 
-    const { depositId, blockNumber, transactionIndex } = deposit;
-    let { depositor, destinationChainId } = deposit;
+    const { blockNumber, transactionIndex } = deposit;
+    let { depositId, depositor, destinationChainId } = deposit;
+    depositId = depositId ?? this.numberOfDeposits;
+    assert(depositId >= this.numberOfDeposits, `${depositId} < ${this.numberOfDeposits}`);
+    this.numberOfDeposits = ++depositId;
```

Comment:
Is `depositId` saved outside of this function? If not, we may want to use `depositId + 1` for readability
```suggestion
    this.numberOfDeposits = depositId + 1;
```